libnest/src/transaction/install.rs

```diff
@@ -52,30 +55,36 @@ impl InstallTransaction {
 
     /// Extracts the downloaded file and performs the installation
     pub fn extract(&self, config: &Config, _: &LockFileOwnership) -> Result<(), Error> {
-        let tarball_path = config
+        let npf_path = config
             .paths()
             .downloaded()
             .join(self.target().repository().as_str())
             .join(self.target().category().as_str())
             .join(self.target().name().as_str())
-            .join(self.target().version().to_string().as_str())
-            .join("data")
-            .with_extension("tar.gz");
-
-        let mut tarball =
-            File::open(&tarball_path).with_context(|_| tarball_path.display().to_string())?;
-        let mut archive = Archive::new(GzDecoder::new(&tarball));
+            .join(format!(
+                "{}-{}.nest",
+                self.target().name(),
+                self.target().version()
+            ));
+
+        let npf_explorer = NPFExplorer::from(self.target().name(), &npf_path)?;
+
+        let tarball_handle = npf_explorer
```

Comment:
Remember the data tarball is optional if it is a virtual package =)

Reply:
Yep, I'll add a TODO for it since I'd rather first merge this PR